src/cmd/sign.js

```diff
@@ -111,27 +126,58 @@ export default function sign(
         log.warn('No extension ID specified (it will be auto-generated)');
       }
 
-      const signingResult = await signAddon({
-        apiKey,
-        apiSecret,
-        apiUrlPrefix,
-        apiProxy,
-        timeout,
-        verbose,
-        id,
-        xpiPath: buildResult.extensionPath,
-        version: manifestData.version,
-        downloadDir: artifactsDir,
-        channel,
-      });
+      let signingResult;
+      if (apiUseAddonSubmissionApi) {
+        if (
+          apiUrlPrefix.endsWith('/api/v3/') || apiUrlPrefix.endsWith('/api/v4')
```

Comment:
That'd be a breaking change to webext, wouldn't it?  (Which is fine, but I was trying hard to avoid any changes to existing users).  We [advise](https://addons-server.readthedocs.io/en/latest/topics/api/overview.html#api-versions) v4 be used, generally, except this api is v5+ so that's not an option.

I'll leave changes decisions about existing functionality to you and @rpl - but it'd be easier to have less checks and more predictable use of api versions, sure.

Reply:
I'm OK with breaking changes behind the `apiUseAddonSubmissionApi` flag that is marked as experimental. So we could start to introduce new CLI flags if we want to (e.g. `apiHost`) and ignore those we don't want to use anymore (e.g. `apiUrlPrefix`).

We could also avoid breaking changes by only using v5 for `apiUseAddonSubmissionApi`.

Reply:
It'd mean a user would have to define by `apiHost` and `apiUrlPrefix` to use a non-prod server if they were using both the existing signing api and the new submission api.  But sure, if you want.